docker_cli/_cli.py

```diff
@@ -0,0 +1,327 @@
+    )
+
+    env_parse = argparse.ArgumentParser(add_help=False)
+    env_parse.add_argument(
+        "--env-file", default="spec-file.txt", type=str,
+        help="The location of the spec-file. Default: \"spec-file.txt\""
+    )
+
+    # Add arguments
+    subparsers = parser.add_subparsers(dest="command")
+
+    setup_parser = subparsers.add_parser("setup", help="Docker image setup commands.",
+                                         formatter_class=help_formatter)
+
+    setup_subparsers = setup_parser.add_subparsers(dest="setup_subcommand")
+
+    setup_all_parser = setup_subparsers.add_parser(
+        "all", parents=[cuda_run_parse, no_cache_parse],
+        help="Set up the full docker image stack.",
+        formatter_class=help_formatter
+    )
+    setup_all_parser.add_argument(
+        "--tag", "-t", default="setup", type=str,
+        help='The sub-prefix of the docker images to be created. Generated images will '
+        f'have tags: "{prefix}-[TAG]-*". Default: "setup"'
+    )
+    setup_all_parser.add_argument(
+        "--base", "-b", default="oraclelinux:8.4", type=str,
+        help='The name of the parent docker image. Default: "oraclelinux:8.4"'
+    )
+    setup_all_parser.add_argument(
+        "--runtime-env-file", default="runtime-spec-file.txt", type=str,
+        help='The location of the runtime spec-file. Default: "runtime-spec-file.txt"'
+    )
+    setup_all_parser.add_argument(
+        "--dev-env-file", default="dev-spec-file.txt", type=str,
+        help='The location of the dev spec-file. Default: "dev-spec-file.txt"'
+    )
+
+    setup_init_parser = setup_subparsers.add_parser(
+        "init", parents=[no_cache_parse],
+        help="Set up the configuration image.",
+        formatter_class=help_formatter
+    )
+    setup_init_parser.add_argument(
+        "--base", "-b", default="oraclelinux:8.4", type=str, required=True,
+        help='The name of the parent docker image. Default: "oraclelinux:8.4"'
+    )
+    _add_tag_argument(parser=setup_init_parser, default="init")
+
+    setup_cuda_parser = setup_subparsers.add_parser(
+        "cuda",
+        help="Set up a CUDA image. Designate dev or runtime.",
+        formatter_class=help_formatter
+    )
+
+    cuda_subparsers = setup_cuda_parser.add_subparsers(dest="cuda_subcommand")
+    setup_cuda_runtime_parser = cuda_subparsers.add_parser(
+        "runtime", parents=[setup_parse, cuda_run_parse, no_cache_parse],
+        help="Set up the CUDA runtime image.",
+        formatter_class=help_formatter
+    )
+    _add_tag_argument(parser=setup_cuda_runtime_parser, default="cuda-runtime")
+
+    setup_cuda_dev_parser = cuda_subparsers.add_parser(
+        "dev", parents=[setup_parse, no_cache_parse],
+        help="Set up the CUDA dev image.",
+        formatter_class=help_formatter
+    )
+    _add_tag_argument(parser=setup_cuda_dev_parser, default="cuda-dev")
+
+    setup_env_parser = setup_subparsers.add_parser(
+        "env",
```

Comment:
I'd recommend calling this something like "conda" instead of "env" to be a little bit more clear/specific about what's being included in this image layer. I think "env" seems a little too vague.

```suggestion
        "conda",
```